internal/controllers/extension_controller.go

```diff
@@ -156,10 +227,161 @@ func (r *ExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ocv1alpha1.Extension{}).
+		Owns(&kappctrlv1alpha1.App{}).
 		Watches(&catalogd.Catalog{}, handler.EnqueueRequestsFromMapFunc(extensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
 		Complete(r)
 }
 
+// TODO: follow up with mapping of all the available App statuses: https://github.com/carvel-dev/kapp-controller/blob/855063edee53315811a13ee8d5df1431ba258ede/pkg/apis/kappctrl/v1alpha1/status.go#L28-L35
+// mapAppStatusToInstalledCondition currently maps only the installed condition.
+func mapAppStatusToInstalledCondition(existingApp *kappctrlv1alpha1.App, ext *ocv1alpha1.Extension, bundleImage string) {
+	appReady := findStatusCondition(existingApp.Status.GenericStatus.Conditions, kappctrlv1alpha1.ReconcileSucceeded)
+	if appReady == nil {
+		ext.Status.InstalledBundleResource = ""
+		setInstalledStatusConditionUnknown(&ext.Status.Conditions, "install status unknown", ext.Generation)
+		return
+	}
+
+	if appReady.Status != corev1.ConditionTrue {
+		ext.Status.InstalledBundleResource = ""
+		setInstalledStatusConditionFailed(
+			&ext.Status.Conditions,
+			appReady.Message,
+			ext.GetGeneration(),
+		)
+		return
+	}
+
+	// InstalledBundleResource this should be converted into a slice as App allows fetching
+	// from multiple sources.
+	ext.Status.InstalledBundleResource = bundleImage
```

Comment:
This is something we should definitely discuss more. There might be a few ways to handle this. Perhaps `InstalledBundleResource` is not the correct thing to capture anyway.

It could be that the correct thing to capture is something a bit more general (e.g. `.name` of the `olm.bundle` blob in the catalog).

This is the kind of thing that would be a good discussion topic for our WG sync or office hours.

Reply:
Right, I'll create a separate issue to follow this up and dig deeper into the use of `InstalledBundledResource` in general. It's definitely good to have, but I'm curious on how it is being consumed externally. Captured in: https://github.com/operator-framework/operator-controller/issues/672